src/js/providers/utils/play-promise.ts

```diff
@@ -33,8 +41,8 @@ export default function createPlayPromise(video) {
     });
 }
 
-function playPromiseError(name, code, message) {
-    const error = new Error(message);
+function playPromiseError(name: string, code: number, message: string): PlayPromiseError {
+    const error = new Error(message) as PlayPromiseError;
```

Comment:
hmmmm - it's tough with the way the code is currently structured but i'm trying to think if there's a quick way to use inheritance here or something similar to avoid the `as`. it's not a deal breaker, i'm fine with this as is but just something to ponder a bit if you'd like

Reply:
This is a polyfill for `media.play(): Promise<void>`
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play
https://html.spec.whatwg.org/multipage/media.html#dom-media-play

The types of errors we are emulating have codes: `NotAllowedError`, `NotSupportedError` of interface`DOMException`...

So I think we shouldn't define our own type - instead use `as DOMException`?

Reply:
Looks like the `code` property is no longer used in favor of just the `name`.
https://developer.mozilla.org/en-US/docs/Web/API/DOMException#Properties